jedi/evaluate/docstrings.py

```diff
@@ -55,14 +59,49 @@ def _search_param_in_numpydocstr(docstr, param_str):
                 m = re.match('([^,]+(,[^,]+)*?)(,[ ]*optional)?$', p_type)
                 if m:
                     p_type = m.group(1)
-
-                if p_type.startswith('{'):
-                    types = set(type(x).__name__ for x in literal_eval(p_type))
-                    return list(types)
-                else:
-                    return [p_type]
+                return _expand_typestr(p_type)
         return []
 
+    def _search_return_in_numpydocstr(docstr):
+        r"""
+        Search `docstr` (in numpydoc format) for type(-s) of `param_str`.
+        """
+        doc = NumpyDocString(docstr)
+        returns = doc._parsed_data['Returns']
```

Comment:
I just realized again that we're using a non-public API here. Can we please have a try/except around it and catch both key error and

```
try:
    # This is a non-public API. If it ever changes we should be prepared and return gracefully.
    returns = doc._parsed_data['Returns']
    returns += doc._parsed_data['Yields']
except (KeyError, AttributeError):
    return []
```

Please also change this for the param version of this function.